packages/ra-ui-materialui/src/input/SelectInput.js

```diff
@@ -155,7 +156,10 @@ export class SelectInput extends Component {
 
     addAllowEmpty = choices => {
         if (this.props.allowEmpty) {
-            return [<MenuItem value="" key="null" />, ...choices];
+            const value = this.props.hasOwnProperty('emptyValue')
+                ? this.props.emptyValue
```

Comment:
Using `defaultProps` might be better than this manual check

Reply:
i agree. will change that later this day